### ISSUE TITLE
[MJAVADOC-517] NPE under Java 10 RC

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#819]
**[Ian Young](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=iay)** opened **[MJAVADOC-517](https://issues.apache.org/jira/browse/MJAVADOC-517?redirect=false)** and commented

Under Java 10 RC, the plugin generates the following exception:

```java
Caused by: java.lang.NullPointerException
    at org.apache.commons.lang3.SystemUtils.isJavaVersionAtLeast (SystemUtils.java:1626)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.getJavadocExecutable (AbstractJavadocMojo.java:3683)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.executeReport (AbstractJavadocMojo.java:2001)
    at org.apache.maven.plugins.javadoc.JavadocJar.doExecute (JavadocJar.java:190)
```

This seems to be down to the version string:

```java
java version "10" 2018-03-20
Java(TM) SE Runtime Environment 18.3 (build 10+43)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10+43, mixed mode)
```

`commons-lang3` version 3.5 can't deal with this, see https://issues.apache.org/jira/browse/LANG-1365 

Inserting an explicit dependency on `commons-lang3` version 3.7 for the plugin works round the issue, and changing the dependency seems like the right fix for the next release of this plugin.

---

**Affects:** 3.0.0

**Issue Links:**
- [MJAVADOC-518](https://issues.apache.org/jira/browse/MJAVADOC-518) Old commons-lang3 dependency causes NPE with JDK 10
 (_**"is duplicated by"**_)
- [MJAVADOC-522](https://issues.apache.org/jira/browse/MJAVADOC-522) Javadoc generation broken on JDK 10 (Commons Lang3 gives NullPointerException)
 (_**"is duplicated by"**_)
- [MJAVADOC-524](https://issues.apache.org/jira/browse/MJAVADOC-524) Running javadoc plugin under JDK 10 fails with NPE
 (_**"is duplicated by"**_)
- [MSITE-822](https://issues.apache.org/jira/browse/MSITE-822) JDK 10: Null Pointer Exception

- [LANG-1365](https://issues.apache.org/jira/browse/LANG-1365) Fix NullPointerException in isJavaVersionAtLeast on Java 10, add SystemUtils.IS_JAVA_10, add JavaVersion.JAVA_10
 (_**"depends upon"**_)

0 votes, 6 watchers

**Comment by jira-importer** [comment]
**[Ian Young](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=iay)** commented

Incidentally, `commons-lang3` seems to have a bug relative to functioning under Java 11 EA even in commons-lang 3.7, so this will come up again in the next few months. I raised a ticket there: https://issues.apache.org/jira/browse/LANG-1384

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [c922342ed87805c236d68ccbf545efb17af0f965](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=c922342ed87805c236d68ccbf545efb17af0f965)
I've replaced usage of SystemUtils from commons-lang3, so we should be good with every new version of Java.
